platform/ui/src/hooks/useSessionStorage.tsx

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect, useCallback } from 'react';
+
+/**
+ * A set of session storage keys that should be cleared out of session storage
+ * when the page unloads.
+ */
+const sessionKeysToClearOnUnload: Set<string> = new Set<string>();
+
+const clearOnUnloadCallback = () => {
+  Array.from(sessionKeysToClearOnUnload.keys()).forEach(key => {
+    window.sessionStorage.removeItem(key);
+  });
+
+  // This is here for unit testing since there is no actual unload between tests
+  // and the set of keys need to be cleared between tests like it would in
+  // a browser between different sessions.
+  sessionKeysToClearOnUnload.clear();
+};
+
+/**
+ * Technically there is no memory leak here because the listener needs to
+ * persist until the page unloads and once the page unloads it will be gone.
+ *
+ * ToDo: unload is not reliably fired on various browsers - particularly mobile.
+ * So on some systems the various session storage items will NOT be cleared
+ * when the page is refreshed or URL altered in some way. Alternatively,
+ * a 'visiblitychange' event could be used whereby on 'hidden' the session item
+ * is cleared but maintained in memory in case a 'visible' event is later fired.
+ */
+window.addEventListener('unload', clearOnUnloadCallback);
```

Comment:
`beforeunload` might be more reliable, but it is NOT meant for clean up like `unload` is meant for and in fact how we are using it.

Yes visibiltychange would address the mobile problem - no need for a combination. But after thinking about it I think we should leave it as `unload`, leave the ToDo see how it goes on mobile (since it is a bit of an edge case) and then come back to it if it is indeed a problem. For the case we are using it, we might get lucky.

Reply:
But if the visibilitychange address mobile why not do it?

Reply:
I suppose I forgot to mention that it is a little bit more work and it is a bit more complicated than what we have here for the unload. 

So we would listen for `hidden` and clear the session storage but ALSO we need to persist the filters in memory in case we get a subsequent `visible` event. A `hidden` event does NOT necessarily mean the page is being unloaded. For example, `hidden` might mean that the user switched to another application or minimized the browser. So if they come back we would get a `visible` event and then we would have to restore the session storage again. It is a bit of a pain for such an edge case.

Reply:
And the `unload` might indeed be good enough for all browsers with how we are using it. I think it is a good first shot at it.

Reply:
Alright, after trying a couple of mobile devices where the unload was not handled, I will implement the `visibilitychange` approach.